moda-contracts/src/Catalog.sol

```diff
@@ -274,25 +277,19 @@ contract Catalog is ICatalog, AccessControlUpgradeable {
         _requireUserIsTrackArtistOrManager(trackId, msg.sender);
         _requireReleasesContractIsRegistered(releases);
         $._singleTrackReleasesPermission[trackId][releases] = hasApproval;
-        if (hasApproval) {
-            emit TrackApprovedToReleases(trackId, releases);
-        } else {
-            emit TrackRemovedFromReleases(trackId, releases);
-        }
+
+        emit TrackApprovalChanged(trackId, releases, hasApproval);
     }
 
     /// @inheritdoc IReleasesApproval
     function setReleasesApprovalForAll(address artist, address releases, bool hasApproval) external {
         CatalogStorage storage $ = _getCatalogStorage();
-
-        _requireUserHasTrackAccess(msg.sender, artist);
+        _requireTrackAccess(msg.sender, artist);
         _requireReleasesContractIsRegistered(releases);
+
         $._allTracksReleasesPermission[artist][releases] = hasApproval;
-        if (hasApproval) {
-            emit AllTracksApprovedToReleases(artist, releases);
-        } else {
-            emit AllTracksRemovedFromReleases(artist, releases);
-        }
+
+        emit AllTracksApprovalChanged(artist, releases, hasApproval);
```

Comment:
This simplifies things a bit and keeps the contract smaller. One event to express both scenarios